### PR TITLE
Cholesky memory test

### PR DESCRIPTION
**Pull request by Sely85**
Added two tests to verify matrix distribution in Cholesky factorization.
In case of local implementation, check if the matrix is actually local.
In case of distributed implementation, check if matrix is distributed according to the given communicator grid.

### CODE REVIEW

File: test/unit/mc/test_cholesky.cpp [hunk 1]
```diff
@@ -182,12 +194,12 @@ TYPED_TEST(CholeskyDistributedTest, NoSquareMatrixException) {
...
 
-TYPED_TEST(CholeskyDistributedTest, NoSquareBlockException) {
+TYPED_TEST(CholeskyDistributedTest, BlockNotSquareException) {
   for (const auto& comm_grid : this->commGrids()) {
     for (const auto& size : square_sizes) {
       for (const auto& block_size : rectangular_block_sizes) {
-        comm::Index2D src_rank_index(std::max(0, comm_grid.size().rows() - 1),
-                                     std::min(1, comm_grid.size().cols() - 1));
+        dlaf::comm::Index2D src_rank_index(std::max(0, comm_grid.size().rows() - 1),
+                                           std::min(1, comm_grid.size().cols() - 1));
```
Comment (rasolca) [inline test/unit/mc/test_cholesky.cpp]: `dlaf::` can be omitted (see using statements in this file)

File: test/unit/mc/test_cholesky.cpp [hunk 2]
```diff
@@ -197,3 +209,37 @@ TYPED_TEST(CholeskyDistributedTest, NoSquareBlockException) {
...
+          Size2D grid_distributed = Size2D(comm_grid.size().rows() + 1, comm_grid.size().cols() + 1);
+          Distribution distribution(sz, block_size, grid_distributed, comm_grid.rank(), {0, 0});
+          Matrix<TypeParam, Device::CPU> mat(std::move(distribution));
+
+          EXPECT_THROW(cholesky(comm_grid, blas::Uplo::Lower, mat), std::invalid_argument);
+        }
+
+        {
+          // Different rank
+          dlaf::comm::Index2D rank_index_distributed(std::max(0, comm_grid.rank().row() - 1),
```
Comment (rasolca) [inline test/unit/mc/test_cholesky.cpp]: Size2D and Index2D are both in namespace `dlaf::comm::`.
Please be consistent in namespace usage (see general comment).

File: test/unit/mc/test_cholesky.cpp [hunk 3]
```diff
@@ -197,3 +209,38 @@ TYPED_TEST(CholeskyDistributedTest, NoSquareBlockException) {
...
+  // Test: matrix not distributed according to the given communicator grid
+  for (const auto& comm_grid : this->commGrids()) {
+    for (const auto& size : square_sizes) {
+      for (const auto& block_size : square_block_sizes) {
+        GlobalElementSize sz = globalTestSize(size);
+
+        {
+          // Different grid size
+          comm::Size2D grid_distributed =
+              comm::Size2D(comm_grid.size().rows() + 1, comm_grid.size().cols() + 1);
```
Comment (rasolca) [inline test/unit/mc/test_cholesky.cpp]: Why you do not use the constructor directly?

**Comment by Sely85** [comment]
Right, sorry. Let me try and fix it

**Comment by Sely85** [comment]
Right, sorry. Let me try and fix it

**Pull request merged**